tests/test_model.py

```diff
@@ -14,66 +14,92 @@
 
 from benchcab import internal
 from benchcab.model import Model, remove_module_lines
+from benchcab.utils.repo import Repo
 
 from .conftest import DEFAULT_STDOUT
 
 
 @pytest.fixture()
-def repo(mock_cwd, mock_subprocess_handler, mock_environment_modules_handler):
+def mock_repo():
+    """Return a mock implementation of the `Repo` interface."""
+
+    class MockRepo(Repo):
+        """A mock implementation of the `Repo` interface used for testing."""
+
+        def __init__(self) -> None:
+            self.handle = "trunk"
+
+        def checkout(self, verbose=False):
+            pass
+
+        def get_branch_name(self) -> str:
+            return self.handle
+
+        def get_revision(self) -> str:
+            pass
+
+    return MockRepo()
+
+
+@pytest.fixture()
+def model(
+    mock_repo, mock_cwd, mock_subprocess_handler, mock_environment_modules_handler
+):
     """Return a mock `Model` instance for testing against."""
-    _repo = Model(path="trunk")
-    _repo.root_dir = mock_cwd
-    _repo.subprocess_handler = mock_subprocess_handler
-    _repo.modules_handler = mock_environment_modules_handler
-    return _repo
+    _model = Model(repo=mock_repo)
+    _model.root_dir = mock_cwd
+    _model.subprocess_handler = mock_subprocess_handler
+    _model.modules_handler = mock_environment_modules_handler
+    return _model
 
 
-class TestRepoID:
-    """Tests for `Model.repo_id`."""
+class TestModelID:
+    """Tests for `Model.model_id`."""
 
-    def test_set_and_get_repo_id(self, repo):
-        """Success case: set and get repository ID."""
+    def test_set_and_get_model_id(self, model):
+        """Success case: set and get model ID."""
         val = 456
-        repo.repo_id = val
-        assert repo.repo_id == val
+        model.model_id = val
+        assert model.model_id == val
 
-    def test_undefined_repo_id(self, repo):
-        """Failure case: access undefined repository ID."""
-        repo.repo_id = None
+    def test_undefined_model_id(self, model):
+        """Failure case: access undefined model ID."""
+        model.model_id = None
         with pytest.raises(
-            RuntimeError, match="Attempting to access undefined repo ID"
+            RuntimeError, match="Attempting to access undefined model ID"
         ):
-            _ = repo.repo_id
+            _ = model.model_id
 
 
 class TestGetExePath:
-    """Tests for `CableRepository.get_exe_path()`."""
+    """Tests for `Model.get_exe_path()`."""
 
-    def test_serial_exe_path(self, repo, mock_cwd):
+    def test_serial_exe_path(self, model, mock_cwd):
         """Success case: get path to serial executable."""
         assert (
-            repo.get_exe_path()
-            == mock_cwd / internal.SRC_DIR / repo.name / "offline" / internal.CABLE_EXE
+            model.get_exe_path()
+            == mock_cwd / internal.SRC_DIR / model.name / "offline" / internal.CABLE_EXE
         )
 
 
+@pytest.mark.skip(reason="The `checkout` method was removed.")
```

Comment:
Why are we keeping these tests here at all?

Reply:
I think I didn't remove it because I haven't implemented the tests for the repo module. I can add a comment here to remove these tests once the tests are implemented.